TestLargeBatch is flaky

```
  000008:[b#11,SET-b#11,SET]
    db_test.go:433: 
        	Error Trace:	/Users/radu/go/src/github.com/cockroachdb/pebble/db_test.go:433
        	Error:      	Received unexpected error:
        	            	expected L0.0:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble.TestLargeBatch.TestLargeBatch.func1.func5
        	            	  | 	/Users/radu/go/src/github.com/cockroachdb/pebble/db_test.go:378
        	            	  | github.com/cockroachdb/pebble.try
        	            	  | 	/Users/radu/go/src/github.com/cockroachdb/pebble/db_test.go:48
        	            	  | github.com/cockroachdb/pebble.TestLargeBatch
        	            	  | 	/Users/radu/go/src/github.com/cockroachdb/pebble/db_test.go:433
        	            	  | testing.tRunner
        	            	  | 	/Users/radu/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.0.darwin-arm64/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	/Users/radu/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.0.darwin-arm64/src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) expected L0.0:
        	            	  |   000005:[a#10,SET-a#10,SET]
        	            	  |   000007:[b#11,SET-b#11,SET]
        	            	  | , but found L0.0:
        	            	  |   000005:[a#10,SET-a#10,SET]
        	            	  |   000008:[b#11,SET-b#11,SET]
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestLargeBatch
--- FAIL: TestLargeBatch (26.25s)
```

Jira issue: PEBBLE-449

--- a/db_test.go
+++ b/db_test.go
@@ -366,12 +366,13 @@ func TestLargeBatch(t *testing.T) {
 	}))
 	require.NoError(t, err)
 
-	verifyLSM := func(expected string) func() error {
+	verifyLSM := func(expectedRegex string) func() error {
+		expected := regexp.MustCompile(expectedRegex)
 		return func() error {
 			d.mu.Lock()
 			s := d.mu.versions.currentVersion().String()
 			d.mu.Unlock()
-			if expected != s {
+			if !expected.MatchString(s) {
 				if testing.Verbose() {
 					fmt.Println(strings.TrimSpace(s))
 				}
@@ -425,13 +426,13 @@ func TestLargeBatch(t *testing.T) {
 
 	// Verify this results in one L0 table being created.
 	require.NoError(t, try(100*time.Microsecond, 20*time.Second,
-		verifyLSM("L0.0:\n  000005:[a#10,SET-a#10,SET]\n")))
+		verifyLSM("L0.0:\n  \\d+:\\[a#10,SET-a#10,SET\\]\n")))
 
 	require.NoError(t, d.Set([]byte("b"), bytes.Repeat([]byte("b"), 512), nil))
 
 	// Verify this results in a second L0 table being created.
 	require.NoError(t, try(100*time.Microsecond, 20*time.Second,
-		verifyLSM("L0.0:\n  000005:[a#10,SET-a#10,SET]\n  000007:[b#11,SET-b#11,SET]\n")))
+		verifyLSM("L0.0:\n  \\d+:\\[a#10,SET-a#10,SET\\]\n  \\d+:\\[b#11,SET-b#11,SET\\]\n")))
 
 	// Allocate a bunch of batches to exhaust the batchPool. None of these
 	// batches should have a non-zero count.

--- a/objstorage/objstorageprovider/vfs_readable.go
+++ b/objstorage/objstorageprovider/vfs_readable.go
@@ -8,6 +8,7 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"runtime/debug"
 	"sync"
 
 	"github.com/cockroachdb/pebble/internal/invariants"
@@ -50,9 +51,10 @@ func newFileReadable(
 		readaheadConfig: readaheadConfig,
 	}
 	if invariants.UseFinalizers {
+		stack := debug.Stack()
 		invariants.SetFinalizer(r, func(obj interface{}) {
 			if obj.(*fileReadable).file != nil {
-				fmt.Fprintf(os.Stderr, "Readable was not closed")
+				fmt.Fprintf(os.Stderr, "Readable was not closed\n%s", stack)
 				os.Exit(1)
 			}
 		})

